Servers/UI/OJS.Servers.Ui/ClientApp/src/components/contests/contest/Contest.tsx

```diff
@@ -98,34 +97,70 @@ const Contest = () => {
         [],
     );
 
-    return (
-        <>
-            <div className={styles.headingContest}>
-                <Heading
-                  type={HeadingType.primary}
-                  className={styles.contestHeading}
-                >
-                    {contest?.name}
-                </Heading>
-                <Heading type={HeadingType.secondary} className={secondaryHeadingClassName}>
-                    {renderTimeRemaining()}
-                    {renderScore()}
-                </Heading>
-            </div>
-
-            <div className={styles.contestWrapper}>
-                <div className={navigationContestClassName}>
-                    <ContestTasksNavigation />
-                </div>
-                <div className={submissionBoxClassName}>
-                    <SubmissionBox />
-                </div>
-                <div className={problemInfoClassName}>
-                    <ContestProblemDetails />
+    const errorMessage = useMemo(
+        () => !validation.contestIsFound
+            ? `${contest?.name} - Contest not found!`
+            : validation.contestIsExpired
+                ? `${contest?.name} - Contest expired!`
+                : !validation.isParticipantRegistered
+                    ? `${contest?.name} - You are not registered for this contest!`
+                    : !validation.contestCanBeCompeted || !validation.contestCanBePracticed
+                        ? `${contest?.name} - You can not take part in the contest!`
+                        : null,
+        [ validation, contest ],
+    );
+
+    const renderErrorMessage = useCallback(
```

Comment:
The naming is incorrect. 
This does not render an `errorMessage`. It renders an error massage in one case and the contest in the other.

Reply:
Renamed to renderPage.